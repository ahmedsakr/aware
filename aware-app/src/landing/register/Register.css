#register {
    background-color: #232f3e;
    border-radius: 7px;
    padding: 20px;
}

#register > h2, #login > h4 {
    color: #c1c7c9;
    text-align: center;    
}

#register-username, #register-password, #register-confirm {
    margin-top: 15px;
}

#register-login {
    margin-top: 10px;
}