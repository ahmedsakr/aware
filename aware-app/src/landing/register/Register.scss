/*
 * Register styling
 *
 * Defines layout fundamentals for the register page.
 */

$background_color: #232f3e;
$font_color: #c1c7c9;

#register {
    background-color: $background_color;
    border-radius: 7px;
    padding: 20px;

    >h2 {
        color: $font_color;
        text-align: center;
    }
}

#login>h4 {
    color: $background_color;
    text-align: center;
}

#register-username,
#register-password,
#register-confirm {
    margin-top: 15px;
}

#register-login {
    margin-top: 10px;
}