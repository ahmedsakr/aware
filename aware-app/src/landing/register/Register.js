import React, { Component } from 'react';
import './Register.css'

class Register extends Component {
    render() {
        return(
            <div id="register">
                <h2 id="welcome-message">Create an Aware Account</h2>
                <form>
                    <div class="container">
                        <label id="register-username">Username</label>
                        <input id="landing-textfield" type="text"></input>

                        <label id="register-password">Password</label>
                        <input id="landing-textfield" type="password"></input>

                        <label id="register-confirm">Confirm Password</label>
                        <input id="landing-textfield" type="password"></input>

                        <button type="submit">Register</button>

                        <label id="register-login">Already have an account? <a href="#" onClick={() => this.props.switch()}>login now!</a></label>
                    </div>
                </form>
            </div>
        );
    }
}

export default Register;
