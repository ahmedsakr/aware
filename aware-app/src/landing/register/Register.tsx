import React from 'react';
import * as verification from '../../shared/verification/user';
import './Register.scss'

type RegisterProps = {
    setUsername: (username: string) => void,
    switch: () => void,
    socket: SocketIOClient.Socket 
};

type RegisterState = {
    username: string,
    password: string,
    confirmPassword: string
};

export default class Register extends React.Component<RegisterProps, RegisterState> {

    constructor(props: RegisterProps) {
        super(props);

        this.state = {
            username: "",
            password: "",
            confirmPassword: ""
        };

        this.handleChange = this.handleChange.bind(this);
    }

    handleChange(event: React.ChangeEvent<HTMLInputElement>): void {
        this.setState({
            [event.target.name]: event.target.value
        } as any);
    }

    register(): void {
        if (!verification.verifyUsername(this.state.username)) {
            alert("Please provide a username between 3 and 32 characters.");
            return;
        }

        if (!verification.verifyPassword(this.state.password)) {
            alert("Please provide a password between 8 and 128 characters.");
            return;
        }

        if (this.state.password === this.state.confirmPassword) {
            this.props.socket.emit('register', this.state.username, this.state.password);
        } else {
            alert("Passwords do not match!");
        }
    }

    render() {
        return(
            <div id="register">
                <h2 id="welcome-message">Create an Aware Account</h2>
                    <div className="container">
                        <label id="register-username">Username</label>
                        <input
                            className="landing-textfield"
                            name="username"
                            type="text"
                            value={this.state.username}
                            onChange={this.handleChange} />

                        <label id="register-password">Password</label>
                        <input
                            className="landing-textfield"
                            name="password"
                            type="password"
                            value={this.state.password}
                            onChange={this.handleChange} />

                        <label id="register-confirm">Confirm Password</label>
                        <input
                            className="landing-textfield"
                            name="confirmPassword"
                            type="password"
                            value={this.state.confirmPassword}
                            onChange={this.handleChange} />

                        <button type="submit" onClick={() => {this.register()}}>Register</button>
                        <label id="register-login">Already have an account? <span onClick={() => this.props.switch()}>login now!</span></label>
                    </div>
            </div>
        );
    }
}
