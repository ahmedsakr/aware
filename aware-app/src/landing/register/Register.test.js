import React from 'react';
import { shallow } from 'enzyme';

import Register from './Register';

describe('Register', () => {
  it('should render correctly in "debug" mode', () => {

    const component = shallow(<Register debug />);
  
    expect(component).toMatchSnapshot();
  });
});