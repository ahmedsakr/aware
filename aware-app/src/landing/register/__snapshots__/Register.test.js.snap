// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register should render correctly in "debug" mode 1`] = `
<div
  id="register"
>
  <h2
    id="welcome-message"
  >
    Create an Aware Account
  </h2>
  <form>
    <div
      class="container"
    >
      <label
        id="register-username"
      >
        Username
      </label>
      <input
        class="landing-textfield"
        type="text"
      />
      <label
        id="register-password"
      >
        Password
      </label>
      <input
        class="landing-textfield"
        type="password"
      />
      <label
        id="register-confirm"
      >
        Confirm Password
      </label>
      <input
        class="landing-textfield"
        type="password"
      />
      <button
        type="submit"
      >
        Register
      </button>
      <label
        id="register-login"
      >
        Already have an account? 
        <span
          onClick={[Function]}
        >
          login now!
        </span>
      </label>
    </div>
  </form>
</div>
`;
